### PR TITLE
fix(griffin): make aggregate check case-insensitive

### PR DESCRIPTION
**Pull request by bziobrowski**
Ref: #2272 fixes performance degradation in group-by queries, caused by SQLOptimiser being case-sensitive to function names. For example this query will be slow (3.5s)

```sql
select x, SUM(double) from tab
```

whereas this query will be fast (0.15s)

```sql
select x, sum(double) from tab
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|

**Review by ideoma — COMMENTED** [review]
Tests please

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 204 / 256 (79.69%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/FullBwdDataFrameCursorFactory.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/StrColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/CountVectorAggregateFunction.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/table/VirtualRecordCursorFactory.java|0|7|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/SymbolColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoIntColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/BooleanColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cairo/sql/Function.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/Long256Column.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/BinColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/RecordColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cairo/IntervalBwdDataFrameCursorFactory.java|0|4|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoByteColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoShortColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/GeoLongColumn.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/cast/CastTimestampToIntFunctionFactory.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/cairo/sql/RecordCursorFactory.java|0|2|00.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/CountGroupByFunction.java|1|3|33.33%|
|:large_blue_circle:|io/questdb/griffin/engine/table/DataFrameRecordCursorFactory.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/griffin/PlanSink.java|39|47|82.98%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/DateColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MaxDoubleGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/CharColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MaxDateVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/constants/LongConstant.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgDoubleVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/IntervalFwdDataFrameCursorFactory.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/SumLongGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/date/AddLongToTimestampFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MaxDoubleVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MaxLongVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByNotKeyedVectorRecordCursorFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/date/NowFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinTimestampGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/FullFwdDataFrameCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MaxFloatGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/KSumDoubleGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinIntVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/FloatColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinTimestampVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinFloatGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/SumDoubleVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MaxCharGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/math/MulIntFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MaxIntVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinDoubleGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MaxLongGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/ShortColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByRecordCursorFactory.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/AvgDoubleGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/math/SubTimestampFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/NSumDoubleVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/SumLongVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinDateGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgLongVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/SumIntVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MaxTimestampGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/AvgIntVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinDoubleVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MaxDateGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinLongGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/DoubleColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/GroupByRecordCursorFactory.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/SumTimestampVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinIntGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinLongVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/std/LongList.java|9|9|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/SumDoubleGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/date/TimestampAddFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/math/MulLongFunctionFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/model/RuntimeIntervalModel.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/AbstractDataFrameCursorFactory.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MinCharGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/LongColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/KSumDoubleVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/SumIntGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/SumDateVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/NSumDoubleGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/TimestampColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MaxTimestampVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/GroupByNotKeyedRecordCursorFactory.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/constants/IntConstant.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/SumFloatGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/ByteColumn.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/groupby/vect/MinDateVectorAggregateFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/groupby/MaxIntGroupByFunction.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/engine/functions/columns/IntColumn.java|2|2|100.00%|

File: core/src/main/java/io/questdb/griffin/engine/functions/cast/CastTimestampToIntFunctionFactory.java [hunk 1]
```diff
@@ -63,5 +64,11 @@ public int getInt(Record rec) {
             final long value = arg.getTimestamp(rec);
             return value == Numbers.LONG_NaN ? Numbers.INT_NaN : (int) value;
         }
+
+        @Override
+        public void toSink(CharSink sink) {
+            sink.put("CastTimestampToInt(").put(arg).put(")");
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/cast/CastTimestampToIntFunctionFactory.java:70]:
```suggestion
            sink.put("CastTimestampToInt(").put(arg).put(')');
```

File: core/src/main/java/io/questdb/griffin/engine/functions/columns/TimestampColumn.java [hunk 2]
```diff
@@ -56,6 +57,11 @@ public boolean isReadThreadSafe() {
         return true;
     }
 
+    @Override
+    public void toSink(CharSink sink) {
+        sink.put("TimestampColumn(").put(columnIndex).put(")");
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/columns/TimestampColumn.java:62]:
```suggestion
        sink.put("TimestampColumn(").put(columnIndex).put(')');
```

Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/engine/functions/columns/TimestampColumn.java:62]: there are more i'm sure

File: core/src/main/java/io/questdb/cairo/FullBwdDataFrameCursorFactory.java [hunk 3]
```diff
@@ -47,4 +48,10 @@ public DataFrameCursor getCursor(SqlExecutionContext executionContext, int order
     public int getOrder() {
         return ORDER_DESC;
     }
-}
\ No newline at end of file
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("FullBwdDataFrame");
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cairo/FullBwdDataFrameCursorFactory.java:54]: Can we use reflection perhaps to output the class name except `CursorFactory` by default?

File: core/src/main/java/io/questdb/griffin/engine/functions/columns/ByteColumn.java [hunk 4]
```diff
@@ -57,6 +58,11 @@ public boolean isReadThreadSafe() {
         return true;
     }
 
+    @Override
+    public void toSink(CharSink sink) {
+        sink.put("ByteColumn");
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/engine/functions/columns/ByteColumn.java:63]: minor: I'd add at least index here, ideally column name

File: core/src/main/java/io/questdb/cairo/sql/Function.java [hunk 5]
```diff
@@ -171,4 +172,8 @@ default boolean supportsRandomAccess() {
 
     default void toTop() {
     }
+
+    default void toSink(CharSink sink) {
+        throw new UnsupportedOperationException("Unssuported for class" + getClass());
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cairo/sql/Function.java:177]: default implementation with reflected class name would be better

File: core/src/main/java/io/questdb/griffin/engine/functions/columns/ByteColumn.java (hunk 4 above)
Comment (bziobrowski) [reply core/src/main/java/io/questdb/griffin/engine/functions/columns/ByteColumn.java]: Added index because class doesn't have column name info .

**Review by ideoma — APPROVED** [review]

**Pull request merged**